pulp_rpm/app/advisory.py

```diff
@@ -77,16 +75,13 @@ def resolve_advisories(version, previous_version):
         # diff for querysets works fine but the result is not fully functional queryset,
         # e.g. filtering doesn't work
         added_advisories = current_advisories.difference(previous_advisories)
-
-    current_advisories_by_id = defaultdict(list)
-    for advisory in current_advisories:
-        current_advisories_by_id[advisory.id].append(advisory)
-
-    added_advisories_by_id = current_advisories_by_id
-    if current_advisories != added_advisories:
         added_advisories_by_id = defaultdict(list)
         for advisory in added_advisories:
             added_advisories_by_id[advisory.id].append(advisory)
+    else:
+        previous_advisory_ids = set()
+        added_advisories = current_advisories
+        added_advisories_by_id = current_advisories_by_id
```

Comment:
The logic feels easier to understand when the values are defined in this if / else block rather than defined and maybe-or-maybe-not modified.